test/src/test_trmm_out.cpp

```diff
@@ -0,0 +1,189 @@
+    real_t bImag = real_t(valueB * (float)rand() / (float)RAND_MAX);
+
+    setScalar(alpha, aReal, aImag);
+    setScalar(beta, bReal, bImag);
+
+    const Uplo uplo = GENERATE(Uplo::Lower, Uplo::Upper);
+
+    const Side side = GENERATE(Side::Left, Side::Right);
+
+    const Op transB = GENERATE(Op::NoTrans, Op::ConjTrans, Op::Trans);
+
+    const Diag diag = GENERATE(Diag::NonUnit, Diag::Unit);
+
+    const Op transA = GENERATE(Op::NoTrans);
+
+    DYNAMIC_SECTION("n = " << n << " m = " << m << " alpha = " << alpha
+                           << " beta = " << beta << " Uplo = " << uplo
+                           << " transB = " << transB)
+    {
+        const real_t eps = ulp<real_t>();
+        const real_t tol = real_t(n) * eps;
+
+        idx_t nc;
+        idx_t mc;
+        idx_t ma;
+
+        if (transB == Op::NoTrans) {
+            nc = n;
+            mc = m;
+        }
+        else {
+            nc = m;
+            mc = n;
+        }
+
+        std::vector<T> B_;
+        auto B = new_matrix(B_, m, n);
+
+        std::vector<T> C_;
+        auto C = new_matrix(C_, mc, nc);
+
+        std::vector<T> C_copy_;
+        auto C_copy = new_matrix(C_copy_, mc, nc);
+
+        if (side == Side::Right) {
+            if (transB == Op::NoTrans) {
+                ma = n;
+            }
+            else {
+                ma = m;
+            }
+        }
+        else {
+            if (transB == Op::NoTrans) {
+                ma = m;
+            }
+            else {
+                ma = n;
+            }
+        }
+
+        std::vector<T> A_;
+        auto A = new_matrix(A_, ma, ma);
+
+        std::vector<T> A_copy_;
+        auto A_copy = new_matrix(A_copy_, ma, ma);
+
+        MatrixMarket mm;
+
+        mm.random(A);
+        mm.random(B);
+        mm.random(C);
+
+        for (idx_t j = 0; j < ma; ++j) {
+            for (idx_t i = 0; i < ma; ++i) {
+                A_copy(i, j) = T(0);
+            }
+        }
+
+        lacpy(Uplo::General, C, C_copy);
+        lacpy(uplo, A, A_copy);
+
+        if (diag == Diag::Unit) {
+            for (idx_t i = 0; i < ma; ++i) {
+                A_copy(i, i) = T(1);
+            }
+        }
+
+        real_t normCbefore = lange(Norm::Fro, C_copy);
+        real_t normA = lantr(Norm::Fro, uplo, diag, A);
+        real_t normB = lange(Norm::Fro, B);
+
+        trmm_out(side, uplo, transA, diag, transB, alpha, A, B, beta, C);
+
+        if (side == Side::Right)
+            gemm(transB, transA, alpha, B, A_copy, beta, C_copy);
+        else {
+            gemm(transA, transB, alpha, A_copy, B, beta, C_copy);
+        }
+
+        for (idx_t j = 0; j < nc; ++j) {
+            for (idx_t i = 0; i < mc; ++i) {
+                C(i, j) -= C_copy(i, j);
+            }
+        }
+
+        real_t normC = lange(Norm::Fro, C);
+
+        normC = normC /
+                ((real(alpha) * normA * normB) + (real(beta) * normCbefore));
+
+        CHECK(normC <= tol);
```

Comment:
With respect to CHECK(normC <= tol), the issue is that if the check fails and the precision is 1e-16 then we see something like 0.0 <= 0.0 because the precision of print of Catch2 is not much.

As of 2.8.0 there is a new way to do this but @luis-carlos-gutierrez reported that 2.8.0 does not work in the CI. For reference:
https://github.com/catchorg/Catch2/blob/devel/docs/tostring.md#floating-point-precision
and then something like:
        Catch::StringMaker<float>::precision = 15;
works great. (But not in the CI.)

One way to go around this is simply to write:
if ( !( normC <= tol ) ){ print normC and to to some accuracy }